### PR TITLE
Refactored Taste Selection

### PR DESCRIPTION
**Pull request by bgouws**
### Refactored Taste Selection

Taste selection now follows MVVM/MVP

### Breakdown

- Removed old dead code 

- Taste Selection follows MVVM/MVP

### CODE REVIEW

File: PTFramework/Classes/MusicTaste/MusicTasteViewModelType.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Brandon Gouws on 2020/04/24.
+//
+
+import Foundation
+
+public protocol MusicTasteViewModelType {
+    var view: MusicTasteViewType? { get set }
+    var repo: MusicTasteRepoType? { get set }
+    init()
```
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteViewModelType.swift:13]: Codacy Issue found: [Lines should not have trailing whitespace.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

**Comment by bgouws** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           

Clones added
============
- PTFramework/Classes/MusicTaste/MusicTasteRepo.swift  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258&bid=17757799)

File: PTFramework/Classes/MusicTaste/MusicTasteViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import Foundation
+
+public class MusicTasteViewModel: MusicTasteViewModelType {
+    public weak var view: MusicTasteViewType?
+    public var repo: MusicTasteRepoType?
+    required public init() { }
+    public func storeTaste(tasteList: [String]) {
+        repo?.StoreTaste(taste: tasteList, completion: { (result) in
+            switch (result) {
```
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteViewModel.swift:16]: Codacy Issue found: [`if`, `for`, `guard`, `switch`, `while`, and `catch` statements shouldn't unnecessarily wrap their conditionals or arguments in parentheses.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

File: PTFramework/Classes/MusicTaste/MusicTasteRepo.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+//  Created by Brandon Gouws on 2020/04/24.
+//
+
+import Foundation
+import FirebaseDatabase
+import FirebaseAuth
+
+public class MusicTasteRepo: MusicTasteRepoType {
+    public required init() { }
+    public func StoreTaste(taste: [String], completion: @escaping(Result<Bool, Error>) -> Void) {
```
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:14]: Codacy Issue found: [Function name should start with a lowercase character: 'StoreTaste(taste:completion:)'](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

File: PTFramework/Classes/MusicTaste/MusicTasteRepo.swift [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+        }
+        ref.child("users").child(userID).child("Tasks").child("Task1").setValue(["Hour": "00",
+                                                                                 "Minute": "01",
+                                                                                 "Second": "30",
+                                                                                 "Title": "Sample"]) {
+            (error: Error?, ref:DatabaseReference) in
```
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:39]: Codacy Issue found: [Colons should be next to the identifier when specifying a type and next to the key in dictionary literals.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:39]: Codacy Issue found: [Closure parameters should be on the same line as opening brace.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:39]: Codacy Issue found: [Unused parameter "ref" in a closure should be replaced with _.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

File: PTFramework/Classes/MusicTaste/MusicTasteRepoType.swift [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+//  PTFramework
+//
+//  Created by Brandon Gouws on 2020/04/24.
+//
+
+import Foundation
+
+public protocol MusicTasteRepoType {
+    init()
+    func StoreTaste(taste: [String], completion: @escaping(Result<Bool, Error>) -> Void)
```
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepoType.swift:12]: Codacy Issue found: [Function name should start with a lowercase character: 'StoreTaste(taste:completion:)'](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

**Comment by bgouws** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Clones added
============
- PTFramework/Classes/MusicTaste/MusicTasteRepo.swift  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258&bid=17757799)

File: PTFramework/Classes/MusicTaste/MusicTasteRepo.swift [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+        }
+        ref.child("users").child(userID).child("Details").setValue(["bio": "Tell us something about yourself!",
+                                                                    "firstName": "Put your name here!",
+                                                                    "lastName": "Put your last name here!"]) {
+            (error: Error?, ref:DatabaseReference) in
```
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:29]: Codacy Issue found: [Closure parameters should be on the same line as opening brace.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:29]: Codacy Issue found: [Unused parameter "ref" in a closure should be replaced with _.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

File: PTFramework/Classes/MusicTaste/MusicTasteRepo.swift [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+        }
+        ref.child("users").child(userID).child("Tasks").child("Task1").setValue(["Hour": "00",
+                                                                                 "Minute": "01",
+                                                                                 "Second": "30",
+                                                                                 "Title": "Sample"]) {
+            (error: Error?, _: DatabaseReference) in
```
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:39]: Codacy Issue found: [Closure parameters should be on the same line as opening brace.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

File: PTFramework/Classes/MusicTaste/MusicTasteRepo.swift (hunk 6 above)
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:29]: Codacy Issue found: [Colons should be next to the identifier when specifying a type and next to the key in dictionary literals.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

File: PTFramework/Classes/MusicTaste/MusicTasteViewModelType.swift (hunk 1 above)
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteViewModelType.swift:13]: Codacy Issue found: [Lines should not have trailing whitespace.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

**Comment by bgouws** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Clones added
============
- PTFramework/Classes/MusicTaste/MusicTasteRepo.swift  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258&bid=17757799)

File: PTFramework/Classes/MusicTaste/MusicTasteRepo.swift [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+        }
+        ref.child("users").child(userID).child("Tasks").child("Task1").setValue(["Hour": "00",
+                                                                                 "Minute": "01",
+                                                                                 "Second": "30",
+                                                                                 "Title": "Sample"])
+        { (error: Error?, _: DatabaseReference) in
```
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:39]: Codacy Issue found: [Opening braces should be preceded by a single space and on the same line as the declaration.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

File: PTFramework/Classes/MusicTaste/MusicTasteRepo.swift [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        { (error: Error?, ref: DatabaseReference) in
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+        }
+        ref.child("users").child(userID).child("Details").setValue(["bio": "Tell us something about yourself!",
+                                                                    "firstName": "Put your name here!",
+                                                                    "lastName": "Put your last name here!"])
+        { (error: Error?, ref: DatabaseReference) in
```
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:29]: Codacy Issue found: [Opening braces should be preceded by a single space and on the same line as the declaration.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:29]: Codacy Issue found: [Unused parameter "ref" in a closure should be replaced with _.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

File: PTFramework/Classes/MusicTaste/MusicTasteViewModelType.swift (hunk 1 above)
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteViewModelType.swift:13]: Codacy Issue found: [Lines should not have trailing whitespace.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

**Comment by bgouws** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones added
============
- PTFramework/Classes/MusicTaste/MusicTasteRepo.swift  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258&bid=17757799)

File: PTFramework/Classes/MusicTaste/MusicTasteViewModelType.swift (hunk 1 above)
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteViewModelType.swift:13]: Codacy Issue found: [Lines should not have trailing whitespace.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

File: PTFramework/Classes/MusicTaste/MusicTasteRepo.swift [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+        }
+        ref.child("users").child(userID).child("Tasks").child("Task1").setValue(["Hour": "00",
+                                                                                 "Minute": "01",
+                                                                                 "Second": "30",
+                                                                                 "Title": "Sample"])
+        { (error: Error?, _: DatabaseReference) in
```
Comment (bgouws) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:39]: Codacy Issue found: [Opening braces should be preceded by a single space and on the same line as the declaration.](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258)

**Comment by bgouws** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- PTFramework/Classes/MusicTaste/MusicTasteRepo.swift  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bgouws/PTPod/pullRequest?prid=5416258&bid=17757799)

File: PTFramework/Classes/MusicTaste/MusicTasteRepo.swift [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+import Foundation
+import FirebaseDatabase
+import FirebaseAuth
+
+public class MusicTasteRepo: MusicTasteRepoType {
+    public required init() { }
+    public func storeTaste(taste: [String], completion: @escaping(Result<Bool, Error>) -> Void) {
+        let ref = Database.database().reference()
+        guard let userID = Auth.auth().currentUser?.uid else {
+            return
```
Comment (meirrosendorff) [inline PTFramework/Classes/MusicTaste/MusicTasteRepo.swift:17]: Maybe consider calling a completion(.failure()) so that this can be handled higher up/

**Pull request merged**